src/spikeinterface/curation/model_based_curation.py

```diff
@@ -0,0 +1,430 @@
+                raise ValueError("Labels in predictions do not match those in label_conversion")
+            predictions = [label_conversion[label] for label in predictions]
+
+        classified_units = pd.DataFrame(
+            zip(predictions, probabilities), columns=["prediction", "probability"], index=self.sorting_analyzer.unit_ids
+        )
+
+        # Set predictions and probability as sorting properties
+        self.sorting_analyzer.sorting.set_property("classifier_label", predictions)
+        self.sorting_analyzer.sorting.set_property("classifier_probability", probabilities)
+
+        if export_to_phy:
+            self._export_to_phy(classified_units)
+
+        return classified_units
+
+    def _get_metrics_for_classification(self):
+        """Check if all required metrics are present and return a DataFrame of metrics for classification"""
+
+        import pandas as pd
+
+        quality_metrics, template_metrics = try_to_get_metrics_from_analyzer(self.sorting_analyzer)
+
+        # Create DataFrame of all metrics and reorder columns to match the model
+        calculated_metrics = pd.concat([quality_metrics, template_metrics], axis=1)
+
+        # Remove any metrics for non-existent units, raise error if no units are present
+        calculated_metrics = calculated_metrics.loc[
+            calculated_metrics.index.isin(self.sorting_analyzer.sorting.get_unit_ids())
+        ]
+        if calculated_metrics.shape[0] == 0:
+            raise ValueError("No units present in sorting data")
+
+        # Check all the required metrics have been calculated
+        required_metrics = set(self.required_metrics)
+        if required_metrics.issubset(set(calculated_metrics)):
+            input_data = calculated_metrics[self.required_metrics]
+        else:
+            raise ValueError(
+                "Input data does not contain all required metrics for classification",
+                f"Missing metrics: {required_metrics.difference(calculated_metrics)}",
+            )
+
+        return input_data
+
+    def _check_params_for_classification(self, enforce_metric_params=False, model_info=None):
+        """
+        Check that quality and template metrics parameters match those used to train the model
+
+        Parameters
+        ----------
+        enforce_metric_params : bool, default: False
+            If True and the parameters used to compute the metrics in `sorting_analyzer` are different than the parmeters
+            used to compute the metrics used to train the model, this function will raise an error. Otherwise, a warning is raised.
+        model_info_path : str or Path, default: None
+            Path to model_info.json provenance file
+        """
+
+        quality_metrics_extension = self.sorting_analyzer.get_extension("quality_metrics")
+        template_metrics_extension = self.sorting_analyzer.get_extension("template_metrics")
+
+        if quality_metrics_extension is not None:
```

Comment:
Here we're checking the *parameters* used to compute the metrics. We do this seperately because people might want to be strict (we will use the exact same parameters as were used when training the model) or be more free.

Note: if we unify the template and quality metric extensions, this could be refactored nicely.